azext_iot/dps/providers/device_registration.py

```diff
@@ -0,0 +1,258 @@
+            compute_key=compute_key,
+            certificate_file=certificate_file,
+            key_file=key_file,
+            passphrase=passphrase
+        )
+
+        # Retrieve sdk.runtime_registration
+        self.sdk = self._get_dps_device_sdk()
+
+    def _validate_attestation_params(
+        self,
+        enrollment_group_id: str = None,
+        device_symmetric_key: str = None,
+        compute_key: bool = False,
+        certificate_file: str = None,
+        key_file: str = None,
+        passphrase: str = None,
+    ):
+        from azure.iot.device import X509
+        if compute_key and not enrollment_group_id:
+            raise RequiredArgumentMissingError(COMPUTE_KEY_ERROR)
+
+        self.device_symmetric_key = None
+        self.certificate = None
+        if device_symmetric_key:
+            if not compute_key:
+                self.device_symmetric_key = device_symmetric_key
+            else:
+                self.device_symmetric_key = iot_dps_compute_device_key(
+                    cmd=self.cmd,
+                    registration_id=self.registration_id,
+                    enrollment_id=enrollment_group_id,
+                    symmetric_key=device_symmetric_key,
+                    dps_name=self.dps_name,
+                    resource_group_name=self.resource_group_name,
+                    login=self.login,
+                    auth_type_dataplane=self.auth_type_dataplane,
+                )
+        elif certificate_file or key_file:
+            self.certificate = X509(
+                cert_file=certificate_file,
+                key_file=key_file,
+                pass_phrase=passphrase or ""
+            )
+        elif certificate_file or key_file:
+            raise RequiredArgumentMissingError(CERTIFICATE_FILE_ERROR)
+        # Retrieve the attestation if nothing is provided.
+        else:
+            self._get_attestation_params(enrollment_group_id=enrollment_group_id)
+
+    def _get_idscope(self) -> str:
+        discovery = DPSDiscovery(self.cmd)
+        target = discovery.get_target(
+            self.dps_name,
+            self.resource_group_name,
+            login=self.login,
+            auth_type=self.auth_type_dataplane,
+        )
+        if target.get("idscope"):
+            return target["idscope"]
+        # If cstring is used, will need to retrieve the id scope manually
+        dps_name = target['entity'].split(".")[0]
+        return discovery.get_id_scope(resource_name=dps_name)
+
+    def _get_dps_device_sdk(
+        self
+    ):
+        from azure.iot.device import ProvisioningDeviceClient
+        if self.device_symmetric_key:
+            return ProvisioningDeviceClient.create_from_symmetric_key(
+                provisioning_host=IOTDPS_PROVISIONING_HOST,
+                registration_id=self.registration_id,
+                id_scope=self.id_scope,
+                symmetric_key=self.device_symmetric_key,
+            )
+        elif self.certificate:
```

Comment:
If none of these conditions are met None is returned from _get_dps_device_sdk , is there a risk here?

Reply:
The user can have either certificate or symmetric key auth. The case in which neither certificate or symm key is used should be caught by validate_attestation_params but I will add an extra check/raise error here.